Sources/tuist/TuistApp.swift

```diff
@@ -13,7 +13,13 @@ private enum TuistServer {
             try? ProcessEnv.setVar(Constants.EnvironmentVariables.verbose, value: "true")
         }
 
-        TuistSupport.LogOutput.bootstrap()
+        if (CommandLine.arguments.count > 1 && CommandLine.arguments[1] == "dump") || CommandLine.arguments
+            .contains("--json")
+        {
+            TuistSupport.LogOutput.bootstrap(config: LoggingConfig(loggerType: .json, verbose: false))
+        } else {
+            TuistSupport.LogOutput.bootstrap()
+        }
```

Comment:
I'd consider refactoring the code to make the code more robust by:
- Avoiding hardcoding strings here.
- Moving the information about whether a command is machine-readable or not closer to the command.

Something along the lines of:

```swift
let machineReadableOutput = TuistCommand.configuration
  .subcommands
  .flatMap {  $0.isMachineReadable }
  .map { configuration.name }
  .first { $0.name == CommandLine.arguments.first } != nil || CommandLine.arguments.contains("--json")
```

Reply:
This is definitely elegant, but not that easy to implement. The `TuistCommand.configuration` model is owner by the Swift Argument Parser, so we can't extend it like that.

I also don't think it's that necessary to move `isMachineReadable` to the command since it's very unlikely the list will grow longer.

I agree it's not ideal to hardcode the name of the command, so I did change that.